replication/binlogsyncer.go

```diff
@@ -733,33 +733,53 @@ func (b *BinlogSyncer) parseEvent(s *BinlogStreamer, data []byte) error {
 		// Some events like FormatDescriptionEvent return 0, ignore.
 		b.nextPos.Pos = e.Header.LogPos
 	}
+
+	getCurrentGtidSet := func() GTIDSet {
+		if b.currGset == nil {
+			return nil
+		}
+		return b.currGset.Clone()
+	}
+
+	advanceCurrentGtidSet := func(gtid string) error {
+		if b.currGset == nil {
```

Comment:
initially `currGset ` is nil, while `prevGset` is set
purpose of `b.currGset = b.prevGset.Clone()` to once allocate memory for `currGset` so it would be safe to call `b.currGset.Update ` on line 749